tick/hawkes/inference/tests/hawkes_cumulant_matching_test.py

```diff
@@ -8,212 +8,305 @@
-                                -0.308883], [-0.299366, 0.272764, -0.347764],
-                               [0.48448, 0.331059, 1.591587]]
 
-            np.testing.assert_array_almost_equal(learner.solution,
-                                                 expected_R_pred)
-
-            expected_baseline = [32.788801, 29.324684, -13.275885]
-
-            np.testing.assert_array_almost_equal(learner.baseline,
-                                                 expected_baseline)
-
-            expected_adjacency = [[-2.925945, -5.54899, -1.97438],
-                                  [-2.201373, -5.009153,
-                                   -1.740234], [1.652958, 2.939054, 1.334677]]
-
-            np.testing.assert_array_almost_equal(learner.adjacency,
-                                                 expected_adjacency)
-
-            np.testing.assert_array_almost_equal(
-                learner.objective(learner.adjacency), 149061.5590630687)
-
-            np.testing.assert_array_almost_equal(
-                learner.objective(R=learner.solution), 149061.5590630687)
-
-        def test_hawkes_cumulants_solve_l2(self):
-            """...Test that hawkes cumulant reached expected value with l2
-            penalization
-            """
-            timestamps, baseline, adjacency = Test.get_train_data(decay=3.)
-            learner = HawkesCumulantMatching(
-                100., cs_ratio=0.9, max_iter=300, print_every=30, step=1e-2,
-                solver='adam', penalty='l2', C=0.1, tol=1e-5)
-            learner.fit(timestamps)
-
-            expected_R_pred = [[0.516135, -0.484529,
-                                -0.323191], [-0.265853, 0.291741, -0.35285],
-                               [0.482819, 0.331344, 1.591535]]
-
-            np.testing.assert_array_almost_equal(learner.solution,
-                                                 expected_R_pred)
-
-            expected_baseline = [17.066997, 17.79795, -6.07811]
-
-            np.testing.assert_array_almost_equal(learner.baseline,
-                                                 expected_baseline)
-
-            expected_adjacency = [[-1.310854, -2.640152, -1.054596],
-                                  [-1.004887, -2.886297,
-                                   -1.065671], [0.910245, 1.610029, 0.913469]]
-
-            np.testing.assert_array_almost_equal(learner.adjacency,
-                                                 expected_adjacency)
-
-            np.testing.assert_array_almost_equal(
-                learner.objective(learner.adjacency), 149232.94041039888)
+from tick.hawkes import (
+    HawkesCumulantMatching,
+    HawkesCumulantMatchingTf,
+    HawkesCumulantMatchingPyT
+)
+from tick.hawkes.inference.hawkes_cumulant_matching import HawkesTheoreticalCumulant
+
+
+class Test(InferenceTest):
+    def setUp(self):
+        self.dim = 2
+        np.random.seed(320982)
+
+    @staticmethod
+    def get_simulated_model():
```

Comment:
@Mbompr This is the idea that I was suggesting in the conversation of this PR. I simulate a hawkes process with exponential kernels and I train `HawkesCumulantMatching` on five samples. Loading pickled data would be replaced by this, if you agree.
More importantly, the assertion of almost equality now are between the exact theoretical values from the simulated hawkes process and the calibrated coefficients of `HawkesCumulantMatching`. When these assertion succeed, they should guarantee approximate exactness of algorithm.